### PR TITLE
KBC-1455 Delete branch workspaces when deleting branch

### PR DESCRIPTION
**Pull request by jirkasemmler**
Jira: KBC-1455
Connection: https://github.com/keboola/connection/pull/3185

A test which creates branch, then a workspace and then it deletes the branch. The workspace should be also deleted but it cannot be tested/validated

### CODE REVIEW

**Comment by jirkasemmler** [comment]
aby testy fungovaly, tak je potreba zapnout routy branch workspacu https://github.com/keboola/connection/pull/3188/files#diff-8b80fdc5057f0085562f534bd7bca760fcb3dbde56104fb5b03e6f00bc2c0e6aR39

**Comment by fejjo** [comment]
👁️

File: composer.json [hunk 1]
```diff
@@ -35,8 +35,9 @@
 		"phpcompatibility/php-compatibility": "*",
 		"phpstan/phpstan-shim": "^0.9.2",
 		"keboola/table-backend-utils": "^0.7",
-		"brianium/paratest": "2.*"
-	},
+		"brianium/paratest": "2.*",
+        "ext-pdo": "*"
```
Comment (fejjo) [inline composer.json:39]: toto tu imho nemusi byt nie ? Len si tu metodu presunul do traity tak to uz fungovalo nie ?

File: tests/Common/DevBranchesTest.php [hunk 2]
```diff
@@ -290,6 +294,44 @@ public function testReadOnlyRoleBranchesPermissions()
...
+        // create workspace
+        $workspaceBranchApi = new Workspaces($this->getBranchAwareDefaultClient($branchData['id']));
+        $workspace = $workspaceBranchApi->createWorkspace();
+        $connection = $workspace['connection'];
+
+        // delete branch and it should also delete the workspace
+        $branchesApi->deleteBranch($branchData['id']);
+
+        // there is event for deleted branch
+        $this->findLastEvent($this->_client, [
```
Comment (fejjo) [inline tests/Common/DevBranchesTest.php:317]: to ze sa vyhodi eventa pri delete branch uz testujeme `\Keboola\Test\Common\DevBranchesTest::testBranchCreateAndDelete` potom hned testujes ze ta branch tam nieje a wokrspace neexistuje tak to je asi dostacujuce ale kludne to tu nechaj

**Review by fejjo — COMMENTED** [review]
Test sam o sebe testuje imho co ma, pridal som par poznameciek ale to sa asi este upravi ked to bude Erik niekam pridavat do svojho testu + failol build tak ked tak to checkni :)

File: composer.json (hunk 1 above)
Comment (jirkasemmler) [reply composer.json]: fungovalo, jen storm nadava a imho bychom to tam specifikovane mit meli

Comment (fejjo) [reply composer.json]: Hm divne mozno mas nieco inac nastavene v PHPstorme pretoze mne nic nesvieti.

**Comment by fejjo** [comment]
buidl je 🟢  testy mi presli, asi len pak pingni Erika a dohodnite sa kam to pridate.

**Review by ErikZigo — APPROVED** [review]
👍

**Pull request merged**